### ISSUE TITLE
Nil pointer dereference

### ISSUE THREAD

**Issue by ddddddO** [goreleaser/goreleaser#3669]
### What happened?

I tried to release using goreleaser/goreleaser-action@v4 on GitHub Actions and got a panic.
See below for specific logs.

```console
Run goreleaser/goreleaser-action@v4
Downloading https://github.com/goreleaser/goreleaser/releases/download/v1.14.0/goreleaser_Linux_x86_64.tar.gz
Extracting GoReleaser
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/038[59](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:60)883-20b0-4b95-8bdc-4744d1de4185 -f /home/runner/work/_temp/b91883f7-ed78-43bb-80e8-d1dfd0d25715
GoReleaser latest installed successfully
/opt/hostedtoolcache/goreleaser-action/1.14.0/x64/goreleaser release --rm-dist
  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • couldn't find any tags before "v0.0.2"
    • building...                                    commit=8872978d902369b7ac10b08fe3acb0934133[60](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:61)a9 latest tag=v0.0.2
    • running against a shallow clone - check your CI documentation at https://goreleaser.com/ci
  • parsing tag
  • setting defaults
      • DEPRECATED: `archives.rlcp` will be the default soon, check https://goreleaser.com/deprecations#archivesrlcp for more info
      • DEPRECATED: `archives.replacements` should not be used anymore, check https://goreleaser.com/deprecations#archivesreplacements for more info
  • running before hooks
    • running                                        hook=go mod tidy
    • took: 2s
  • checking distribution directory
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/kaisekisan_linux_arm64/kaisekisan
    • building                                       binary=dist/kaisekisan_linux_amd64_v1/kaisekisan
    • building                                       binary=dist/kaisekisan_linux_386/kaisekisan
    • building                                       binary=dist/kaisekisan_darwin_amd64_v1/kaisekisan
    • building                                       binary=dist/kaisekisan_darwin_arm64/kaisekisan
    • building                                       binary=dist/kaisekisan_windows_amd64_v1/kaisekisan.exe
    • building                                       binary=dist/kaisekisan_windows_arm64/kaisekisan.exe
    • building                                       binary=dist/kaisekisan_windows_386/kaisekisan.exe
    • took: 42s
  • generating changelog
    • writing                                        changelog=dist/CHANGELOG.md
  • archives
    • creating                                       archive=dist/kaisekisan_linux_i386.tar.gz
    • creating                                       archive=dist/kaisekisan_linux_arm64.tar.gz
    • creating                                       archive=dist/kaisekisan_darwin_x86_64.tar.gz
    • creating                                       archive=dist/kaisekisan_darwin_arm64.tar.gz
    • creating                                       archive=dist/kaisekisan_windows_x86_64.zip
    • creating                                       archive=dist/kaisekisan_windows_arm64.zip
    • creating                                       archive=dist/kaisekisan_windows_i386.zip
    • creating                                       archive=dist/kaisekisan_linux_x86_64.tar.gz
    • took: 3s
  • calculating checksums
  • homebrew tap formula
    • guessing install to be "bin.install \"kaisekisan\""
    • guessing install to be "bin.install \"kaisekisan\""
    • guessing install to be "bin.install \"kaisekisan\""
    • guessing install to be "bin.install \"kaisekisan\""
    • writing                                        formula=dist/kaisekisan.rb
  • scoop manifests
    • writing                                        manifest=dist/kaisekisan.json
  • publishing
    • scm releases
      • creating or updating release                 repo=ddddddO/kaisekisan tag=v0.0.2
      • release updated                              name=v0.0.2 release-id=87503122 request-id=31C0:82C4:EF57C8:1EDFA1F:63AD5533
      • uploading to release                         file=dist/kaisekisan_linux_arm64.tar.gz name=kaisekisan_linux_arm64.tar.gz
      • uploading to release                         file=dist/kaisekisan_linux_i386.tar.gz name=kaisekisan_linux_i386.tar.gz
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa7110f]

goroutine 282 [running]:
github.com/goreleaser/goreleaser/internal/client.(*githubClient).Upload(0xc000012d58, 0xc000290000, {0xc0007df[62](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:63)0, 0x8}, 0xc0002b6000, 0x0?)
	github.com/goreleaser/goreleaser@v1.14.0/internal/client/github.go:331 +0x1ef
github.com/goreleaser/goreleaser/internal/pipe/release.upload.func1()
	github.com/goreleaser/goreleaser@v1.14.0/internal/pipe/release/release.go:176 +0x214
github.com/goreleaser/goreleaser/internal/pipe/release.upload(0x12?, {0x7f50f77ebaf8?, 0xc000012d58?}, {0xc0007df620?, 0x405a7d?}, 0xc0002b6000)
	github.com/goreleaser/goreleaser@v1.14.0/internal/pipe/release/release.go:188 +0xf3
github.com/goreleaser/goreleaser/internal/pipe/release.doPublish.func1()
	github.com/goreleaser/goreleaser@v1.14.0/internal/pipe/release/release.go:160 +0x31
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75 +0x[64](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:65)
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.0.0-20220722155255-886fb93[71](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:72)eb4/errgroup/errgroup.go:[72](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:73) +0xa5
Error: The process '/opt/hostedtoolcache/goreleaser-action/1.14.0/x[64](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856#step:4:65)/goreleaser' failed with exit code 2
```

### How can we reproduce this?

- [GitHub Actions 's log](https://github.com/ddddddO/kaisekisan/actions/runs/3799388086/jobs/6461904856)
- [.goreleaser](https://github.com/ddddddO/kaisekisan/blob/v0.0.2/.goreleaser.yml)
- [.github/workflows/cd.yaml](https://github.com/ddddddO/kaisekisan/blob/v0.0.2/.github/workflows/cd.yaml)

### goreleaser version

```bash
goreleaser/goreleaser-action@v4
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ddddddO** [comment]
Hi!👋 @caarlos0 
I made [PR](https://github.com/goreleaser/goreleaser/pull/3670), please review it!

**Comment by caarlos0** [comment]
fixed in #3670 

thanks @ddddddO !
